harvest/api/paper.py

```diff
@@ -113,28 +114,40 @@ def fetch_stock_order_status(self, id: int) -> Dict[str, Any]:
 
         qty = ret['quantity']
        
-        # If order has been filled, simulate asset buy/sell
-        if ret['status'] == 'filled':
+        # If order is open, simulate asset buy/sell if possible
+        if ret['status'] == 'open':
             if is_crypto(ret['symbol']): 
                 lst = self.cryptos
             else:
                 lst = self.stocks
 
             pos = next((r for r in lst if r['symbol'] == sym), None)
             if ret['side'] == 'buy':
-                # If asset already exists, buy more. If not, add a new entry
-                if pos == None:
-                    lst.append({
-                        'symbol': sym,
-                        'avg_price': price,
-                        'quantity': qty
-                    })
+                # Check to see if user has enough funds to buy the stock
+                if self.buying_power < price * qty:
+                    warning(f"""Not enough buying power.\n Total price ({price} * {qty}) exceeds buying power {self.buy_power}.\n Reduce purchase quantity or increase buying power.""")
+                # Check to see the price does not exceed the limit price
+                elif ret['limit_price'] < price:
+                    limit_price = ret['limit_price']
+                    info(f'Limit price for {sym} is less than current price ({limit_price} < {price}).')
```

Comment:
Check to see that the user can purchase the stocks and the price is lower than the limit_price.